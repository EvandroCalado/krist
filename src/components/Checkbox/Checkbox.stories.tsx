import { Meta, StoryFn } from '@storybook/react';

import { Checkbox, CheckboxProps } from '.';

export default {
  title: 'Components/Checkbox',
  component: Checkbox,
} as Meta;

export const Default: StoryFn<CheckboxProps> = () => (
  <Checkbox label="checkbox" name="checkbox" />
);
