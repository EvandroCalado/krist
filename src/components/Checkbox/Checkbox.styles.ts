import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    gap: ${theme.spacings.xs};
  `}
`;

export const Label = styled.label`
  ${({ theme }) => css`
    width: 100%;
    font-size: ${theme.font.sizes.md};
    text-transform: capitalize;
    font-size: ${theme.font.sizes.sm};
    padding: ${theme.spacings.xs} ${theme.spacings.sm};
    border-radius: 0.5rem;
    cursor: pointer;

    &:hover {
      background-color: ${theme.colors.secondary};
    }
  `}
`;

export const Input = styled.input`
  ${({ theme }) => css`
    width: 1.5rem;
    height: 1.5rem;
    accent-color: ${theme.colors.primary};
    cursor: pointer;
  `}
`;
