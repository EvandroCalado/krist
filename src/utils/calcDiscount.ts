export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
