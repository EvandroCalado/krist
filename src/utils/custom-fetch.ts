import axios from 'axios';

export const customFetch = axios.create({
  baseURL: import.meta.env.VITE_STRAPI_URL,
});
