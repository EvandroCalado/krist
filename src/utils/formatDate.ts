export const formatDate = (dateString: string): string => {
  const date = new Date(dateString);

  return date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};
