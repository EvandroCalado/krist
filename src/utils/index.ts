export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
