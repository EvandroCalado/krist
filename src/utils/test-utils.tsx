import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
