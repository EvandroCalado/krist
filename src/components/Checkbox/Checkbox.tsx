import { FC, InputHTMLAttributes } from 'react';

import * as S from './Checkbox.styles';

export interface CheckboxProps extends InputHTMLAttributes<HTMLInputElement> {
  label: string;
}

export const Checkbox: FC<CheckboxProps> = ({ label, ...rest }) => {
  return (
    <S.Container>
      <S.Input type="checkbox" id={label} {...rest} />
      <S.Label htmlFor={label}>{label}</S.Label>
    </S.Container>
  );
};
