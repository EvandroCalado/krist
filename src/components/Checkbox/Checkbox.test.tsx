import { screen } from '@testing-library/react';

import { customRender } from 'utils/test-utils';

import { Checkbox } from './Checkbox';

describe('Checkbox', () => {
  it('should render default', () => {
    customRender(<Checkbox label="checkbox" name="checkbox" />);

    const checkbox = screen.getByRole('checkbox');
    const label = screen.getByText('checkbox');

    expect(checkbox).toBeInTheDocument();
    expect(checkbox).toHaveAttribute('name', 'checkbox');
    expect(checkbox).toHaveAttribute('id', 'checkbox');
    expect(label).toBeInTheDocument();
  });

  it('should render with pass attribute', () => {
    customRender(
      <Checkbox label="checkbox" name="checkbox" placeholder="test" />,
    );

    const checkbox = screen.getByRole('checkbox');

    expect(checkbox).toHaveAttribute('placeholder', 'test');
  });

  it('should render a snapshot', () => {
    const { container } = customRender(
      <Checkbox label="checkbox" name="checkbox" placeholder="test" />,
    );

    expect(container).toMatchSnapshot();
  });
});
