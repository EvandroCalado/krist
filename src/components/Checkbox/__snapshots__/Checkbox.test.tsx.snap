// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render a snapshot 1`] = `
.c0 {
  display: flex;
  align-items: center;
  gap: .8rem;
}

.c2 {
  width: 100%;
  font-size: 1.6rem;
  text-transform: capitalize;
  font-size: 1.2rem;
  padding: .8rem 1rem;
  border-radius: 0.5rem;
  cursor: pointer;
}

.c2:hover {
  background-color: #f0f0f0;
}

.c1 {
  width: 1.5rem;
  height: 1.5rem;
  accent-color: #131118;
  cursor: pointer;
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
      id="checkbox"
      name="checkbox"
      placeholder="test"
      type="checkbox"
    />
    <label
      class="c2"
      for="checkbox"
    >
      checkbox
    </label>
  </div>
</div>
`;
